src/Core/ObjectRegistry.php

```diff
@@ -142,7 +142,7 @@ protected function _checkDuplicate(string $name, array $config): void
 
         $failure = null;
         foreach ($config as $key => $value) {
-            if (!array_key_exists($key, $existingConfig)) {
+            if ($key !== 'className' && !array_key_exists($key, $existingConfig)) {
```

Comment:
It would be good to have some test coverage for components here. I added a test for helpers in #17297.

Reply:
You mean https://github.com/cakephp/cakephp/pull/17274

Reply:
I added a test
```php
    /**
     * testConstructClassesWithComponents method
     */
    public function testConstructClassesWithComponentsInPlugins(): void
    {
        $this->loadPlugins(['TestPlugin']);

        $Controller = new TestPluginController(new ServerRequest());
        $Controller->loadComponent('TestPlugin.Other');
        $Controller->loadComponent('TestPlugin.Plugins');

        $this->assertInstanceOf('TestPlugin\Controller\Component\OtherComponent', $Controller->Other);
        $this->assertInstanceOf('TestPlugin\Controller\Component\PluginsComponent', $Controller->Plugins);
    }
```
but it also didnt fail before, so I wonder what a good indicative test case would be.

Reply:
I added a test
```php
    /**
     * testConstructClassesWithComponents method
     */
    public function testConstructClassesWithComponentsInPlugins(): void
    {
        $this->loadPlugins(['TestPlugin']);

        $Controller = new TestPluginController(new ServerRequest());
        $Controller->loadComponent('TestPlugin.Other');
        $Controller->loadComponent('TestPlugin.Plugins');

        $this->assertInstanceOf('TestPlugin\Controller\Component\OtherComponent', $Controller->Other);
        $this->assertInstanceOf('TestPlugin\Controller\Component\PluginsComponent', $Controller->Plugins);
    }
```
but it also didnt fail before, so I wonder what a good indicative test case would be.

Reply:
OK, so the controller/component issue I was able to resolve with this hack for now:
https://github.com/dereuromark/cakephp-tinyauth/commit/7f586f70d67c9f5fd9bb5a27da39c13a5d973f19
It seems to happen when you load plugin components that extend another one and add config on top.
This makes the core logic trip up, and therefore one needs to manually inject the className here to make sure the exception is not trown.